ui/shared/RevisionLinkify.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactLinkify, { linkify } from 'react-linkify';
+
+const revRe = /([a-f\d]{12,40})/;
+
+export default class RevisionLinkify extends React.Component {
+  constructor(props) {
+    super(props);
+
+    linkify.add('rev:', {
+      validate: function (text, pos, self) {
+        const revision = text.slice(pos).split(' ')[0];
+
+        self.re.rev = revRe;
+        if (self.re.rev.test(revision)) {
+          return revision.match(self.re.rev)[0].length;
+        }
+        return 0;
+      },
+      normalize: (match) => {
```

Comment:
Maybe I'm missing something, but doesn't this return 'undefined' when it's called since you're not returning a value?

Reply:
It would, yeah.  But it's not looking for the return value.  This is just a "hook" to let us make last minute changes to it's default way to make links.  We're modifying the ``match`` object that it passed in.  And it then uses that ``match`` object to make the returned ``jsx`` element.